pandas/core/computation/expressions.py

```diff
@@ -80,11 +80,15 @@ def _can_use_numexpr(op, op_str, a, b, dtype_check):
             # check for dtype compatibility
             dtypes = set()
             for o in [a, b]:
-                if hasattr(o, "dtypes") and o.ndim > 1:
-                    s = o.dtypes.value_counts()
-                    if len(s) > 1:
-                        return False
-                    dtypes |= set(s.index.astype(str))
+                if hasattr(o, "dtypes"):
+                    # Series implements dtypes, check for dimension count
+                    if o.ndim > 1:
+                        s = o.dtypes.value_counts()
+                        if len(s) > 1:
+                            return False
+                        dtypes |= set(s.index.astype(str))
+                    else:
+                        dtypes |= {o.dtypes.name}
```

Comment:
This is the Series case, right? Slight preference to use `o.dtype.name` in that case.

Reply:
@TomAugspurger  Refactored to simplify the logic now. Will use dtypes if a Dataframe and dtype if ndarray and Series.